### ISSUE TITLE
Error Assertion version 3.2.0 Infinite Scroll

### ISSUE THREAD

**Issue by Javernaver** [Klerith/animate_do_package#31]
Hi, I have this widget whats previously works fine in the package version 3.1.2. But updating to 3.2.0 is raising exception

This is the widget where is called, happens in the moment to load the next page in the infinite scroll, Maybe I need to update something in the call of FadeInUp now?

```import 'package:animate_do/animate_do.dart';
import 'package:flutter/material.dart';
import 'package:flutter_bloc/flutter_bloc.dart';
import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';

import 'package:bsale_mobile_v2/features/inventory/presentation/views/views.dart';
import 'package:bsale_mobile_v2/features/shared/shared.dart';
import 'package:bsale_mobile_v2/features/inventory/domain/entities/entities.dart';
import 'package:bsale_mobile_v2/features/inventory/presentation/blocs/blocs.dart';
import 'package:bsale_mobile_v2/config/config.dart';

class InventoryHistoryListView extends StatefulWidget {
  const InventoryHistoryListView({super.key});

  @override
  State<InventoryHistoryListView> createState() =>
      _InventoryHistoryListViewState();
}

class _InventoryHistoryListViewState extends State<InventoryHistoryListView> {
  late final InventoryHistoryBloc _inventoryHistoryBloc;

  static const _pageSize = Environment.defaultPageSize;

  final PagingController<int, InventoryHistory> _pagingController =
      PagingController(firstPageKey: 0);

  @override
  void initState() {
    super.initState();
    _inventoryHistoryBloc = context.read<InventoryHistoryBloc>();
    // * Petición inicial y evento del bloc para obtener los datos
    _pagingController.addPageRequestListener((pageKey) {
      // * Evento del bloc para obtener los datos
      _inventoryHistoryBloc.add(LoadInventoryHistory(
        filters: {
          'limit': _pageSize,
          'offset': pageKey,
        },
      ));
    });
  }

  @override
  void dispose() {
    super.dispose();
    _pagingController.dispose();
    _inventoryHistoryBloc.close();
  }

  @override
  Widget build(BuildContext context) {
    return BlocConsumer<InventoryHistoryBloc, InventoryHistoryState>(
      // * IMPORTANTE: para evitar doble append de datos, solo se debe escuchar cuando el estado de status cambie
      listenWhen: (previous, current) => previous.status != current.status,
      listener: (context, state) {
        if (state.status == InventoryHistoryStatus.success) {
          // * si es la primera carga se asigna limpia la lista y se asigna la nueva obtenida (tambien si se filtra desde cero)
          if (state.filters['offset'] == 0) {
            _pagingController.itemList = [];
          }
          // determinar si es la ultima pagina
          final isLastPage = state.inventoryHistoryNews.length < _pageSize;
          // determinar pagina siguiente
          final next = state.inventoryHistory.length;
          if (isLastPage) {
            _pagingController.appendLastPage(state.inventoryHistoryNews);
          } else {
            _pagingController.appendPage(
              state.inventoryHistoryNews,
              next,
            );
          }
        }
        if (state.status == InventoryHistoryStatus.filtering ||
            state.status == InventoryHistoryStatus.reloading) {
          logger.i('filtering & refreshing: ${state.searchTerm}');
          // * se respalda la lista de items actual y se asigna nuevamente para mantener el listado mientras se filtran los datos, ya que el refesh limpia la lista
          final items = _pagingController.itemList;
          _pagingController.refresh();
          _pagingController.itemList = items;
        }
        if (state.status == InventoryHistoryStatus.failure) {
          Utils.showToast(I18n.translate('errors.noLoad'));
        }
      },
      builder: (context, state) {
        return RefreshIndicator(
          onRefresh: () async {
            final inventoryHistoryBloc = context.read<InventoryHistoryBloc>();
            inventoryHistoryBloc.add(const ReloadInventoryHistory());
            // * Espera mediante un stream que el siguiente estado sea success (cuando se complete el pull to refresh)
            await inventoryHistoryBloc.stream.firstWhere(
              (state) =>
                  state.status == InventoryHistoryStatus.success ||
                  state.status == InventoryHistoryStatus.failure,
              orElse: () => inventoryHistoryBloc.state,
            );
          },
          child: PagedListView<int, InventoryHistory>.separated(
            pagingController: _pagingController,
            separatorBuilder: (context, index) => const SizedBox(height: 5),
            builderDelegate: PagedChildBuilderDelegate(
              firstPageProgressIndicatorBuilder: (context) =>
                  const SizedBox.shrink(),
              animateTransitions: true,
              firstPageErrorIndicatorBuilder: (context) {
                return Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: Text(I18n.translate('errors.noLoad')),
                );
              },
              noItemsFoundIndicatorBuilder: (context) {
                return Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: Text(I18n.translate('common.noResults')),
                );
              },
              newPageProgressIndicatorBuilder: (context) => const Padding(
                padding: EdgeInsets.only(bottom: 10),
                child: Center(child: Loader()),
              ),
              itemBuilder: (context, history, index) {
                return FadeInUp(
                  child: InventoryHistoryItemView(inventoryHistory: history),
                );
              },
            ),
          ),
        );
      },
    );
  }
}

```

error ```_AssertionError ('package:flutter/src/animation/animation_controller.dart': Failed assertion: line 453 pos 7: '_ticker != null': AnimationController.forward() called after AnimationController.dispose()
AnimationController methods should not be used after calling dispose.)```

![image](https://github.com/Klerith/animate_do_package/assets/38194420/ee05692c-9f40-4ff1-9d7b-dd7377614e6f)

Flutter doctor
[√] Flutter (Channel stable, 3.16.9, on Microsoft Windows [Versi¢n 10.0.22631.3085], locale es-CL)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2022 17.5.4)
[√] Android Studio (version 2023.1)
[√] VS Code (version 1.85.2)
[√] Connected device (4 available)
[√] Network resources

**Comment by Klerith** [comment]
@Javernaver, I'm not able to recreate the issue. Can you provide me with a simple example where the error pops out?

**Comment by Javernaver** [comment]
Hi, I was able to recreate the exception scrolling up and down a little fast in the examples from [infinite_scroll_pagination](https://pub.dev/packages/infinite_scroll_pagination), and using this examples in this repo adding animations: [https://github.com/Javernaver/fer-test](https://github.com/Javernaver/fer-test)

Maybe is using this package is the problem, or flutter_bloc or flutter itself? 🤔 In 3.1.2 dont happens anyways.
![image](https://github.com/Klerith/animate_do_package/assets/38194420/59580fef-bf38-4862-b46b-f22e746fad0a)
![image](https://github.com/Klerith/animate_do_package/assets/38194420/3887d537-f2e4-4a3a-9848-6d74dcce25aa)

If you cant find anything, just close the issue, maybe I need to refactor how the bloc rebuild the screen in the pagination😄

**Comment by Klerith** [comment]
Thank you!

I was able to reproduce the error; this should be fixed in the next release, 3.3.1

**Comment by Klerith** [comment]
@Javernaver ya lo arregle, la versión 3.3.1 lo corrigió.

Disculpa la molestia.
